scan-command-test-utils/src/main/java/io/ballerina/scan/test/ScanAssertions.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.scan.test;
+
+import io.ballerina.scan.Issue;
+import io.ballerina.scan.Rule;
+import io.ballerina.scan.RuleKind;
+import io.ballerina.scan.Source;
+import org.testng.Assert;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Test utility class to assert the issues found during a scan.
+ *
+ * @since 0.10.0
+ */
+public class ScanAssertions {
```

Comment:
Thinking out loud, do we need the Scan prefix?

Reply:
If we are to remove the `Scan` prefix, the API usage will look like below in a project.

``` java
Assertions.assertHasIssue(issues, "ballerina/io:1", "main.bal", 23, 23, Source.BUILT_IN);
```

This will be used, likely with another test framework like testNG. In testNG, we already have `Assert` and `Assertion`. (In JUnit we already have an `Assertions` class)
Not having the prefix causes over-generalization of what the class does IMO.

Reply:
IMO, that is okay. `io.ballerina.scan.test.ScanAssertions` repeating Scan feels redundant, especially given that there will no other kind of Assertions class within the scan.test package.

Reply:
Renamed with 1b001c2dc732b1c0d7b7f039a8259a257c623d4e